src/compiler/constraints.jl

```diff
@@ -121,40 +178,54 @@ function toqubo_constraint(
         @warn "Infeasible constraint detected"
     end
 
-    z = encode!(Binary(), model, nothing, zero(T), abs(l))
+    z = encode!(model, Binary(), nothing, zero(T), abs(l))
 
     for (ω, c) in expansion(z)
         g[ω] += c
     end
 
+    # Tell the compiler that quadratization is necessary
+    MOI.set(model, QUADRATIZE(), true)
+
     return g^2
 end
 
 function toqubo_constraint(
-    model::VirtualQUBOModel{T},
-    v::MOI.VectorOfVariables,
+    model::VirtualModel{T},
+    x::MOI.VectorOfVariables,
     ::MOI.SOS1{T},
     ::AbstractArchitecture,
 ) where {T}
-    # -*- Special Ordered Set of Type 1: ∑ x <= min x 😄 -*-
+    # Special Ordered Set of Type 1: ∑ x ≤ min x
     g = PBO.PBF{VI,T}()
 
-    for vi in v.variables
-        for (ωi, _) in expansion(MOI.get(model, Source(), vi))
+    for xi in x.variables
+        vi = model.source[xi]
+        
+        # Currently, SOS1 only supports binary variables
+        @assert encoding(vi) isa Mirror
+
+        for (ωi, _) in expansion(vi)
             g[ωi] = one(T)
         end
     end
 
-    z = expansion(encode!(Mirror(), model, nothing))
+    # Slack variable
+    z = expansion(encode!(model, Mirror(), nothing))
 
-    return (g + z - one(T))^2 # one-hot approach
+    # NOTE: Using one-hot approach. Not great, but it works.
+    # IDEA: Use Domain-Wall.
+    #   Approach 1. Use new variables yᵢ such that
+    #       xᵢ = yᵢ₊₁ - yᵢ
+    #   and substitute every occurence of xᵢ across the model.
```

Comment:
Good insight, maybe leave it as an Issue for enhancements such that it does not disappear when this is merged